### PR TITLE
[Java] okhttp-gson content-type fix for */*

### PR DESCRIPTION
**Pull request by lwander**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Finishes work started in #6188 and #6175

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache [hunk 1]
```diff
@@ -608,11 +608,11 @@ public class ApiClient {
      *
      * @param contentTypes The Content-Type array to select from
      * @return The Content-Type header to use. If the given array is empty,
-     *   JSON will be used.
+     *   or matches "any", JSON will be used.
      */
     public String selectHeaderContentType(String[] contentTypes) {
-        if (contentTypes.length == 0) {
-            return "application/json";
+        if (contentTypes.length == 0 || contentTypes[0].equals("*/*")) {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache]: @lwander we've actually implemented the check in `isJsonMime`: https://github.com/lwander/swagger-codegen/blob/a73ee037181933ee9040cc90383f2d4611a49c85/modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache#L580

Comment (lwander) [inline modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache]: Right, once `isJsonMime` returns true, this function will return the value `*/*`, which is not a useful content-type header.

Comment (wing328) [inline modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache]: You're right. We'll revise other generators to apply similar fix.

**Comment by wing328** [comment]
FYI. Java Petstore sample updated via 646c2bc

**Pull request merged**